packages/shorebird_cli/lib/src/commands/preview_command.dart

```diff
@@ -316,6 +321,72 @@ This is only applicable when previewing Android releases.''',
     return ReleasePlatform.values.firstWhere((p) => p.displayName == platform);
   }
 
+  /// Downloads and runs the given [release] of the given [appId] on Windows.
+  Future<int> installAndLaunchWindows({
+    required String appId,
+    required Release release,
+    required DeploymentTrack track,
+  }) async {
+    const platform = ReleasePlatform.windows;
+    late Directory appDirectory;
+    late ReleaseArtifact releaseExeArtifact;
+
+    try {
+      releaseExeArtifact = await codePushClientWrapper.getReleaseArtifact(
+        appId: appId,
+        releaseId: release.id,
+        arch: 'exe',
+        platform: platform,
+      );
+    } on Exception catch (e, s) {
+      logger
+        ..err('Error getting release artifact: $e')
+        ..detail('Stack trace: $s');
+      return ExitCode.software.code;
+    }
+
+    appDirectory = Directory(
+      getArtifactPath(
+        appId: appId,
+        release: release,
+        artifact: releaseExeArtifact,
+        platform: platform,
+        extension: 'exe',
+      ),
+    );
+
+    if (!appDirectory.existsSync()) {
```

Comment:
It's a bit odd that this "cache if needed" behavior is built into the command here, rather than some central caching class, but that can be a later refactor when needed. 🤷

Reply:
This is specifically caching the artifacts for preview so it makes sense for it to be part of the preview command imo 🤷